src/wallet/wallet.cpp

```diff
@@ -2824,7 +2850,9 @@ bool CWallet::AddToWalletIfInvolvingMe(const CTransaction& tx, const CBlock* pbl
         }
 
         // Orchard
-        orchardWallet.AddNotesIfInvolvingMe(tx);
+        if (consensus.NetworkUpgradeActive(nHeight, Consensus::UPGRADE_NU5)) {
+            orchardWallet.AddNotesIfInvolvingMe(nHeight, tx);
```

Comment:
This misinterprets the `nHeight` parameter (I'm sorry it isn't documented 😞). We added `nHeight` because we needed to know the height at which this transaction was being _targeted for inclusion in the chain_ (in order to correctly enforce ZIP 212), not the height it is actually being mined at. That is, if `pblock == nullptr`, this is a transaction from the mempool, and `nHeight` is a synthetic height. Per my earlier comment, this needs to be something like:
```suggestion
            orchardWallet.AddNotesIfInvolvingMe(pblock ? nHeight : std::nullopt, tx);
```